Formula/i/instaloader.rb

```diff
@@ -0,0 +1,44 @@
+class Instaloader < Formula
+  include Language::Python::Virtualenv
+
+  desc "Download media from Instagram"
+  homepage "https://instaloader.github.io/"
+  url "https://files.pythonhosted.org/packages/53/a0/49ded81d0134be2e1c22ae4706c35b74594ed1844bd1e9af703fd310d562/instaloader-4.10.1.tar.gz"
+  sha256 "902cc8b9569ca7437323199c8e55dbdcd15ea5c8e874c7864e84a36dd55f8584"
+  license "MIT"
+
+  depends_on "python-certifi"
+  depends_on "python@3.12"
+
+  resource "charset-normalizer" do
+    url "https://files.pythonhosted.org/packages/63/09/c1bc53dab74b1816a00d8d030de5bf98f724c52c1635e07681d312f20be8/charset-normalizer-3.3.2.tar.gz"
+    sha256 "f30c3cb33b24454a82faecaf01b19c18562b1e89558fb6c56de4d9118a032fd5"
+  end
+
+  resource "idna" do
+    url "https://files.pythonhosted.org/packages/8b/e1/43beb3d38dba6cb420cefa297822eac205a277ab43e5ba5d5c46faf96438/idna-3.4.tar.gz"
+    sha256 "814f528e8dead7d329833b91c5faa87d60bf71824cd12a7530b5526063d02cb4"
+  end
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/9d/be/10918a2eac4ae9f02f6cfe6414b7a155ccd8f7f9d4380d62fd5b955065c3/requests-2.31.0.tar.gz"
+    sha256 "942c5a758f98d790eaed1a29cb6eefc7ffb0d1cf7af05c3d2791656dbd6ad1e1"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/36/dd/a6b232f449e1bc71802a5b7950dc3675d32c6dbc2a1bd6d71f065551adb6/urllib3-2.1.0.tar.gz"
+    sha256 "df7aa8afb0148fa78488e7899b2c59b5f4ffcfa82e6c54ccb9dd37c1d7b52d54"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    outut = shell_output("#{bin}/instaloader --fast-update notexist 2>&1")
```

Comment:
@chenrui333 Unfortunately the user "notexist" does indeed, exist 😃

Reply:
yeah, I know it does exist :) (but it is private page)

Reply:
I actually see something different in my local

```
$ /opt/homebrew/Cellar/instaloader/4.10.1/bin/instaloader --fast-update notexist

HTTP redirect from https://i.instagram.com/api/v1/users/web_profile_info/?username=notexist to https://i.instagram.com/accounts/login/?next=/api/v1/users/web_profile_info/
notexist: Redirected to login page. Use --login.

Errors or warnings occurred:
notexist: Redirected to login page. Use --login.
```

Reply:
actually the page got changed now 🤯 

![image](https://github.com/Homebrew/homebrew-core/assets/1580956/7c08775d-82f4-42a1-9a22-d8e02114e418)

Reply:
might need a better account for the testing :(

```
    Minitest::Assertion: Expected /notexist:\ Redirected\ to\ login\ page\.\ Use\ \-\-login/ to match "Stored ID 24114340 for profile notexist.\nHint: Use --login to download higher-quality versions of pictures.\n[1/1] Downloading profile notexist\nnotexist/2015-08-12_06-04-16_UTC_profile_pic.jpg notexist: --login=USERNAME required.\n\nErrors or warnings occurred:\nnotexist: --login=USERNAME required.\n\n".
```

Reply:
What about the official instagram account?